source/RevitLookup/ViewModels/Visualization/CurveLoopVisualizationViewModel.cs

```diff
@@ -0,0 +1,211 @@
+// RevitLookup.ViewModels/Visualization/CurveLoopVisualizationViewModel.cs
+using System.Collections.Generic;
+using System.Windows.Media;
+using CommunityToolkit.Mvvm.ComponentModel;
+using JetBrains.Annotations;
+using RevitLookup.Abstractions.Services.Presentation;
+using RevitLookup.Abstractions.Services.Settings;
+using RevitLookup.Abstractions.ViewModels.Visualization;
+using RevitLookup.Core.Visualization;
+using RevitLookup.Core.Visualization.Events;
+using Autodesk.Revit.DB;
+using Microsoft.Extensions.Logging;
+using System.Windows;
+using Color = System.Windows.Media.Color;
+
+namespace RevitLookup.ViewModels.Visualization
+{
+    [UsedImplicitly]
+    public sealed partial class CurveLoopVisualizationViewModel : ObservableObject, ICurveLoopVisualizationViewModel
+    {
+        private readonly CurveLoopVisualizationServer _server = new();
+        private readonly ISettingsService _settingsService;
+        private readonly INotificationService _notificationService;
+        private readonly ILogger<CurveLoopVisualizationViewModel> _logger;
+
+        public CurveLoopVisualizationViewModel(
```

Comment:
Use style as in other classes. Use Primary constructor, assign default values to fields instead of properties. Consider the order of C# members, fields should not be below the constructor